src/crimson/os/seastore/cache.cc

```diff
@@ -1012,6 +1012,15 @@ CachedExtentRef Cache::duplicate_for_write(
     i->version++;
     i->state = CachedExtent::extent_state_t::EXIST_MUTATION_PENDING;
     i->last_committed_crc = i->get_crc32c();
+    // deepcopy the buffer of exist clean extent beacuse it shares
+    // buffer with original clean extent.
+    auto bp = i->get_bptr();
+    auto len = bp.length();
+    char *nb = new char[len];
+    bp.copy_out(0, len, nb);
+    auto nbp = ceph::bufferptr(nb, len);
```

Comment:
There is another copy inside the constructor, probably worth using ptr::c_str() to get the raw char array directly from the source bp.